modules/Awards/Awards/PilotHoursAwards.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Modules\Awards\Awards;
+
+use App\Contracts\Award;
+
+/**
+ * All award classes need to extend Award and implement the check() method
+ * This award is based on the original PilotFlightAwards.php file but
+ * checks the Pilots Flight Time (In Minutes).
+ * This award means you can create an award for a pilot that completes any
+ * amount of flight time (In Minutes).
+ *
+ * See: https://docs.phpvms.net/developers/awards
+ */
+class PilotHoursAwards extends Award
+{
+    /**
+     * Set the name of this award class to make it easier to see when
+     * assigning to a specific award
+     *
+     * @var string
+     */
+    public $name = 'Pilot Flight Time';
+
+    /**
+     * The description to show under the parameters field, so the admin knows
+     * what the parameter actually controls. You can leave this blank if there
+     * isn't a parameter.
+     *
+     * @var string
+     */
+    public $param_description = 'Amount of flight time in Minutes at which to give this award';
+
+    /**
+     * If the user has over N minutes of flights, then we can give them this award.
+     *
+     * If no parameter is passed in, just default it to 99999. You should check if there
+     * is a parameter or not. You can call it whatever you want, since that would make
+     * sense with the $param_description.
+     *
+     * @param int|null $flight_minutes The parameters passed in from the UI
+     *
+     * @return bool
+     */
+    public function check($flight_minutes = null): bool
+    {
+        if (!$flight_minutes) {
```

Comment:
You can remove the first `if (!$flight_minutes) {` check, since it will already be checked by `is_int`. If it's null, IMO it's better not to assume a default value. Also add the log message, it helps greatly with debugging and I can guarantee now someone will ask why the award didn't happen.

Add

```php
use Illuminate\Support\Facades\Log;
```

And let's set the log to 

```php
Log::error('PilotHoursAwards: Invalid flight minutes value of "'.$flight_minutes.'"');
```

Which gives me an idea to add another method in the interface to check the passed parameter to ensure that it's valid, then the admin panel can check that.

Reply:
it has some text in the in-line docs that states

     * If no parameter is passed in, just default it to 99999. You should check if there
     * is a parameter or not.

So I guess that can also be removed with the additional level of input validation.

Reply:
Since that's being included by default, someone is less likely to check the code for that note